### PR TITLE
Error-handling crystal.

### PR DESCRIPTION
**Pull request by rpiaggio**
The new version of `crystal` would force us to handle errors (or log them) in effects started from Callbacks resulting from `.runAsync*CB` methods.

~Opening as draft for the moment until the new `crystal` is released, so we can see how code would look like, in case anyone has ideas to improve this.~

### CODE REVIEW

File: common/src/main/scala/explore/components/ConnectionsStatus.scala [hunk 1]
```diff
@@ -54,7 +54,7 @@ object ConnectionsStatus {
   val component = ScalaComponent
     .builder[Props]
     .render(_ =>
-      AppCtx.withCtx { ctx =>
+      AppCtx.runWithCtx { ctx =>
```
Comment (cquiroz) [inline common/src/main/scala/explore/components/ConnectionsStatus.scala:57]: I recall we discussed changing this name to make it more explicit 👍

File: common/src/main/scala/explore/components/UserSelectionForm.scala [hunk 2]
```diff
@@ -62,50 +68,52 @@ object UserSelectionForm {
...
-                onClick = p.login >> p.message.set(none).runAsyncCB
-              ).when(s.showButtons),
-              Button(content = "Continue as Guest",
-                     size = Big,
-                     clazz = ExploreStyles.LoginBoxButton,
-                     onClick = p.guest >> p.message.set(none).runAsyncCB,
-                     icon = Icons.UserAstronaut
-              ).when(s.showButtons),
-              p.message.get.whenDefined(message =>
+        AppCtx.runWithCtx { implicit ctx =>
```
Comment (cquiroz) [inline common/src/main/scala/explore/components/UserSelectionForm.scala:71]: Why do we need the ctx now if we didn't before?

**Review by cquiroz — APPROVED** [review]
Looks good, some minor comments inline

File: common/src/main/scala/explore/components/UserSelectionForm.scala (hunk 2 above)
Comment (toddburnside) [reply common/src/main/scala/explore/components/UserSelectionForm.scala:71]: For the implicit Logger, I believe.

**Review by toddburnside — APPROVED** [review]
LGTM

File: common/src/main/scala/explore/components/UserSelectionForm.scala (hunk 2 above)
Comment (rpiaggio) [reply common/src/main/scala/explore/components/UserSelectionForm.scala:71]: > For the implicit Logger, I believe.

Yes

**Pull request merged**